### DISCUSSION TITLE
Pallas GPU Kernel under (p)jit

### DISCUSSION THREAD

**Discussion by luyug** [jax-ml/jax#19694] [Q&A] [answered]
I implemented a neural network that has some pallas kernels in it. I wonder what will happen to the kernels if I jit the entire model training step on a multi-GPU system? Will they they also be automatically partitioned?

**Comment by sharadmv** [comment, marked as answer]
Pallas kernels won't be autopartitioned, though we are brainstorming APIs to enable adding hooks into the partitioner. The preferred solution is to use `jax.experimental.shard_map`.
